From 77d0ba3dc3e4b57a6df9d7dd30bedac4d29939c8 Mon Sep 17 00:00:00 2001
From: AnNDP <anndp@fsoft.com.vn>
Date: Sun, 10 Jun 2018 15:40:28 +0700
Subject: [PATCH 4/9]
 gst1122-waylandsink-Restrict-the-number-of-buffers-in-the-pool to be allocated

Signed-off-by: AnNDP <anndp@fsoft.com.vn>
---
 ext/wayland/gstwaylandsink.c | 7 +++----
 ext/wayland/wlbuffer.h       | 1 +
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 1ac7d44..dc03631 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -557,7 +557,7 @@ gst_wayland_create_pool (GstWaylandSink * sink, GstCaps * caps)
   pool = gst_video_buffer_pool_new ();
 
   structure = gst_buffer_pool_get_config (pool);
-  gst_buffer_pool_config_set_params (structure, caps, size, 2, 0);
+  gst_buffer_pool_config_set_params (structure, caps, size, GST_WAYLAND_BUFFER_POOL_NUM, GST_WAYLAND_BUFFER_POOL_NUM);
   gst_buffer_pool_config_set_allocator (structure, gst_wl_shm_allocator_get (),
       NULL);
 
@@ -635,7 +635,7 @@ gst_wayland_sink_propose_allocation (GstBaseSink * bsink, GstQuery * query)
     pool = gst_wayland_create_pool (sink, caps);
 
   if (pool) {
-    gst_query_add_allocation_pool (query, pool, sink->video_info.size, 2, 0);
+    gst_query_add_allocation_pool (query, pool, sink->video_info.size, GST_WAYLAND_BUFFER_POOL_NUM, GST_WAYLAND_BUFFER_POOL_NUM);
     g_object_unref (pool);
   }
 
@@ -799,8 +799,7 @@ gst_wayland_sink_show_frame (GstVideoSink * vsink, GstBuffer * buffer)
 
         config = gst_buffer_pool_get_config (sink->pool);
         gst_buffer_pool_config_get_params (config, &caps, NULL, NULL, NULL);
-        gst_buffer_pool_config_set_params (config, caps, sink->video_info.size,
-            2, 0);
+        gst_buffer_pool_config_set_params (config, caps, sink->video_info.size, GST_WAYLAND_BUFFER_POOL_NUM, GST_WAYLAND_BUFFER_POOL_NUM);
 
         /* This is a video pool, it should not fail with basic setings */
         if (!gst_buffer_pool_set_config (sink->pool, config) ||
diff --git a/ext/wayland/wlbuffer.h b/ext/wayland/wlbuffer.h
index cbb50f7..f97ee3a 100644
--- a/ext/wayland/wlbuffer.h
+++ b/ext/wayland/wlbuffer.h
@@ -31,6 +31,7 @@ G_BEGIN_DECLS
 #define GST_WL_BUFFER_CLASS(klass)          (G_TYPE_CHECK_CLASS_CAST ((klass), GST_TYPE_WL_BUFFER, GstWlBufferClass))
 #define GST_IS_WL_BUFFER_CLASS(klass)       (G_TYPE_CHECK_CLASS_TYPE ((klass), GST_TYPE_WL_BUFFER))
 #define GST_WL_BUFFER_GET_CLASS(obj)        (G_TYPE_INSTANCE_GET_CLASS ((obj), GST_TYPE_WL_BUFFER, GstWlBufferClass))
+#define GST_WAYLAND_BUFFER_POOL_NUM 3
 
 typedef struct _GstWlBuffer GstWlBuffer;
 typedef struct _GstWlBufferClass GstWlBufferClass;
-- 
1.9.1

